### PR TITLE
Upgrade from v1.x operator

### LINKED ISSUE #461: Track changes for automatic upgrade path
Seamlessly upgrade operator from v1.x to v2.x for a given applications namespace.

- [ ] Verify resource update works when deployed with v2.
- [ ] #586
- [ ] #587
        - Verify network policies are not redundant
        - Verify monitoring resources are not redundant
- [ ] Test bundle update

### PR DESCRIPTION
**Pull request by VaishnaviHire**
Fixes #461 
## How Has This Been Tested?
Tested in managed evironment
Dev Build: [quay.io/vhire/opendatahub-operator-catalog:v2.4.0](http://quay.io/vhire/opendatahub-operator-catalog:v2.4.0)

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -192,29 +233,90 @@ func main() {
...
+	if err != nil {
+		return fmt.Errorf("failed to get platform: %v", err)
 	}
+	fmt.Printf("platform is name: %v", platform)
 
-	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
+	if platform != deploy.OpenDataHub && platform != deploy.ManagedRhods {
```
Comment (zdtsw) [inline main.go:263]: so we want to create DSC (only kserve disabled) if it is selfmanaged?

Comment (VaishnaviHire) [reply main.go:263]: No I need to remove the deploy.OpenDataHub from the condition, but essentially return without creating DSC only when platform is Managed.

For Self-managed and ODH we do not create the Default DSC

**Comment by VaishnaviHire** [comment]
@zdtsw This is ready for review again

**Comment by VaishnaviHire** [comment]
Local e2e test run 
```
 make e2e-test 
go test ./tests/e2e/ -run ^TestOdhOperator -v --operator-namespace=openshift-operators "--skip-deletion=false" -timeout 15m 
=== RUN   TestOdhOperator
=== RUN   TestOdhOperator/validate_operator_pod_is_running
=== RUN   TestOdhOperator/validate_operator_pod_is_running/validate_ODH_Operator_pod
=== RUN   TestOdhOperator/validate_operator_pod_is_running/validate_CRDs_owned_by_the_operator
=== RUN   TestOdhOperator/create_Opendatahub_components
I1020 14:56:47.071407   59294 request.go:682] Waited for 1.025287524s due to client-side throttling, not priority and fairness, request: GET:https://api.etirelli.dev.datahub.redhat.com:6443/apis/security.openshift.io/v1?timeout=32s
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Creation_of_DataScienceCluster_instance
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_TrustyAI
=== PAUSE TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_TrustyAI
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_TrustyAI
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve
=== CONT  TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Ownerrefrences_exist
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Controller_reconcile
=== RUN   TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Component_Enabled_field
=== RUN   TestOdhOperator/delete_components
=== RUN   TestOdhOperator/delete_components/e2e-test
=== RUN   TestOdhOperator/delete_components/e2e-test/Deletion:_DataScienceCluster_instance
=== RUN   TestOdhOperator/delete_components/e2e-test/Deletion:_Application_Resource
--- PASS: TestOdhOperator (354.33s)
    --- PASS: TestOdhOperator/validate_operator_pod_is_running (34.59s)
        --- PASS: TestOdhOperator/validate_operator_pod_is_running/validate_ODH_Operator_pod (10.06s)
        --- PASS: TestOdhOperator/validate_operator_pod_is_running/validate_CRDs_owned_by_the_operator (20.16s)
    --- PASS: TestOdhOperator/create_Opendatahub_components (234.91s)
        --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test (230.73s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Creation_of_DataScienceCluster_instance (10.14s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components (60.06s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Ray (10.09s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_DataSciencePipelines (10.09s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_TrustyAI (10.10s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Workbenches (10.13s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Kserve (10.14s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_Dashboard (10.14s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_CodeFlare (120.04s)
                --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_all_deployed_components/Validate_ModelMeshServing (120.04s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Ownerrefrences_exist (0.08s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Controller_reconcile (20.14s)
            --- PASS: TestOdhOperator/create_Opendatahub_components/e2e-test/Validate_Component_Enabled_field (20.27s)
    --- PASS: TestOdhOperator/delete_components (84.83s)
        --- PASS: TestOdhOperator/delete_components/e2e-test (80.47s)
            --- PASS: TestOdhOperator/delete_components/e2e-test/Deletion:_DataScienceCluster_instance (0.11s)
            --- PASS: TestOdhOperator/delete_components/e2e-test/Deletion:_Application_Resource (80.35s)
PASS
ok      github.com/opendatahub-io/opendatahub-operator/v2/tests/e2e     355.232s

```

**Comment by bartoszmajsak** [comment]
@VaishnaviHire @zdtsw This assumes that "legacy" stuff is present on the cluster. What if we start from the clean cluster and only want to use v2 resources? At this point, everything fails to reconcile for me with:

```
2023-10-25T06:26:17Z    ERROR    controllers.DSCInitialization    unable to update from legacy operator version    {"error": "error getting list of kfdefs: no matches for kind \"KfDef\" in version \"kfdef.apps.kubeflow.org/v1\""}
```

**Pull request merged**